### PR TITLE
Add support for installing GPU drivers on Ubuntu

### PR DESCRIPTION
**Pull request by medb**

### CODE REVIEW

File: gpu/install_gpu_driver.sh [hunk 1]
```diff
@@ -20,12 +20,20 @@ set -euxo pipefail
 readonly GPU_AGENT_REPO_URL='https://raw.githubusercontent.com/GoogleCloudPlatform/ml-on-gcp/master/dlvm/gcp-gpu-utilization-metrics'
 
 # Whether to install GPU monitoring agent that sends GPU metrics to StackDriver
-INSTALL_GPU_AGENT="$(/usr/share/google/get_metadata_value attributes/install_gpu_agent || true)"
+INSTALL_GPU_AGENT=$(get_metadata_attribute 'install_gpu_agent' 'false')
```
Comment (functicons) [inline gpu/install_gpu_driver.sh]: I assume this will fail as get_metadata_attribute is not yet defined.

File: gpu/install_gpu_driver.sh [hunk 2]
```diff
@@ -37,48 +45,61 @@ function install_gpu_driver() {
...
+      for distro in ${OS_DIST} ${OS_DIST}-backports; do
+        echo "${type} http://deb.debian.org/debian ${distro} contrib non-free" \
           >>/etc/apt/sources.list.d/non-free.list
       done
     done
-  done
+    packages+=(nvidia-driver nvidia-kernel-common nvidia-smi)
+    modules+=(nvidia-current)
+    local nvblas_cpu_blas_lib=/usr/lib/libblas.so
+  else
```
Comment (functicons) [inline gpu/install_gpu_driver.sh]: It is better to check Ubuntu explicitly, we might have other OS distros in the future.

File: gpu/install_gpu_driver.sh (hunk 1 above)
Comment (medb) [reply gpu/install_gpu_driver.sh]: Fixed

File: gpu/install_gpu_driver.sh (hunk 2 above)
Comment (medb) [reply gpu/install_gpu_driver.sh]: Done.

File: gpu/install_gpu_driver.sh [hunk 3]
```diff
@@ -37,48 +44,63 @@ function install_gpu_driver() {
...
           >>/etc/apt/sources.list.d/non-free.list
       done
     done
-  done
+    packages+=(nvidia-driver nvidia-kernel-common nvidia-smi)
+    modules+=(nvidia-current)
+    local nvblas_cpu_blas_lib=/usr/lib/libblas.so
+  fi
+
+  if [[ ${OS_NAME} == ubuntu ]]; then
```
Comment (functicons) [inline gpu/install_gpu_driver.sh]: If the OS is neither Ubuntu nor Debian, nvblas_cpu_blas_lib won't be initialized, it is better to return an error early.

Comment (medb) [reply gpu/install_gpu_driver.sh]: Done.

**Pull request merged**